youtube_dl/extractor/americastestkitchen.py

```diff
@@ -90,3 +91,69 @@ def _real_extract(self, url):
             'series': try_get(episode, lambda x: x['show']['title']),
             'episode': episode.get('title'),
         }
+
+
+class AmericasTestKitchenSeasonIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?(?P<show>americastestkitchen|cookscountry)\.com/episodes/browse/season_(?P<id>\d+)'
+    _TESTS = [{
+        # ATK Season
+        'url': 'https://www.americastestkitchen.com/episodes/browse/season_1',
+        'info_dict': {
+            'id': 'season-1',
+            'title': 'Season 1',
+        },
+        'playlist_count': 13,
+    }, {
+        # Cooks Country Season
+        'url': 'https://www.cookscountry.com/episodes/browse/season_12',
+        'info_dict': {
+            'id': 'season-12',
+            'title': 'Season 12',
+        },
+        'playlist_count': 13,
+    }, {
+        # Multi-digit season
+        'url': 'https://www.americastestkitchen.com/episodes/browse/season_20',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        show_name, season = re.match(self._VALID_URL, url).groups()
+
+        slug = 'atk' if show_name == 'americastestkitchen' else 'cco'
+
+        filters = [
+            'search_season_list:Season %s' % season,
+            'search_document_klass:episode',
+            'search_show_slug:%s' % slug,
+        ]
+
+        season_search = self._download_json(
+            'https://y1fnzxui30-dsn.algolia.net/1/indexes/everest_search_atk_season_desc_production',
+            season, headers={
+                'Origin': 'https://www.%s.com' % show_name,
+                'X-Algolia-API-Key': '8d504d0099ed27c1b73708d22871d805',
+                'X-Algolia-Application-Id': 'Y1FNZXUI30',
+            }, query={
+                'facetFilters': json.dumps(filters),
+                'attributesToRetrieve': 'search_url',
+                'attributesToHighlight': '',
+                # ATK and CCO generally have less than 26 episodes per season
+                'hitsPerPage': '100',
+            })
+
+        entries = [
+            self.url_result(
+                'https://www.%s.com%s' % (show_name, episode.get('search_url')),
```

Comment:
will return invalid URL if `search_url` is `null`.

Reply:
I've updated the condition to handle this case